### DISCUSSION TITLE
Can Http2 SendRequest send multiple requests in a row?

### DISCUSSION THREAD

**Discussion by OldMansKid** [hyperium/hyper#3703] [Q&A] [answered]
I noticed that the future returned by SendRequest::send_request doesn't depend on the SendRequest instance. I wonder if I can use a single http2 SendRequest to send multiple requests and then use a FuturesUnordered to poll these futures.

**Comment by seanmonstar** [comment, marked as answer]
Yes they can. You can check the various [readiness methods](https://docs.rs/hyper/latest/hyper/client/conn/http2/struct.SendRequest.html#method.ready) to see if you can send another request before calling `send_request()`.

**Reply by OldMansKid** [reply]
Sounds great, thanks for the tip.
